docs/articles_en/documentation/openvino-extensibility/transformation-api/patterns-python-api.rst

```diff
@@ -0,0 +1,130 @@
+.. {#openvino_docs_Extensibility_UG_patterns-python-api}
+
+Overview of creating pattern using Python API
+=============================================
+
+.. meta::
+   :description: Learn how to apply additional model optimizations or transform
+                 unsupported subgraphs and operations, using OpenVINO™ Transformations API.
+
+Pattern matching is an essential component of OpenVINO™ transformations. Before performing any transformation on a subgraph of a graph, we need to find the subgraph in the graph.
+Here come patterns which serve as a searching utility to identify nodes we are going to work with in our transformation. In this article we are going to review the basics of pattern
+creation using Python API and helping utilities we may use to facilitate working with them. Some examples will be intentionally simplified for ease of understanding. 
+
+Though, before proceeding any further, we need to add some imports. That would import the operations we're going to use and additional utility that we are going to talk later in this guide.
+Add the following lines to your file:
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: cpp
+   :fragment: [ov:imports]
+
+This should be enough for our article.
+
+What is a pattern and how to create one?
+++++++++++++++++++++++++++++++++++++++++
+
+Let's start with a very brief definition of pattern. In a nutshell, a pattern is a very simplified model comprised of nodes we want to match. It lacks some features of a model and generally cannot serve as one,
+but this is not very important right now.
+
+Suppose, we are having a very simple pattern consisting of 3 nodes and we want to find it in a given model.
+
+.. image:: ./../../../_static/images/python-api.png
```

Comment:
please add Result operation to the graph in the picture

Reply:
I intentionally did add all the fancy stuff to avoid complexity. Do you think this is important here?

Reply:
Added.